crates/typed-store/tests/macro_tests.rs

```diff
@@ -259,3 +261,16 @@ async fn store_iter_and_filter_successfully() {
     }
     assert_eq!(output.len(), key_values.len());
 }
+
+#[tokio::test]
+async fn test_sampling() {
```

Comment:
Should we also have unit test for non-zero sampling interval?
You can use tokio test with start_paused=true and manual advanced time(sui codebase have examples)

Reply:
oh yeah that's a good point, we should have that test. I;ll add it